src/frontend/data/queries/updateOne.spec.tsx

```diff
@@ -0,0 +1,145 @@
+        'Content-Type': 'application/json',
+      },
+      method: 'PUT',
+      body: JSON.stringify(objectToUpdate),
+    });
+    expect(response).toEqual({ key: 'value' });
+  });
+
+  it('updates the resource without JWT token', async () => {
+    mockAppData = {};
+    const objectToUpdate = { object: 'data' };
+    fetchMock.mock('/api/model-name/1/', { key: 'value' });
+
+    const response = await updateOne({
+      name: 'model-name',
+      id: '1',
+      object: objectToUpdate,
+    });
+
+    expect(fetchMock.lastCall()![0]).toEqual('/api/model-name/1/');
+    expect(fetchMock.lastCall()![1]).toEqual({
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'PUT',
+      body: JSON.stringify(objectToUpdate),
+    });
+    expect(response).toEqual({ key: 'value' });
+  });
+
+  it('resolves with a failure and handles it when it fails to update the resource (local)', async () => {
+    mockAppData = { jwt: 'some token' };
+    const objectToUpdate = { object: 'data' };
+    fetchMock.mock(
+      '/api/model-name/1/',
+      Promise.reject(new Error('Failed to perform the request')),
+    );
+
+    await expect(
+      updateOne({
+        name: 'model-name',
+        id: '1',
+        object: objectToUpdate,
+      }),
+    ).rejects.toThrowError('Failed to perform the request');
+
+    expect(fetchMock.lastCall()![0]).toEqual('/api/model-name/1/');
+    expect(fetchMock.lastCall()![1]).toEqual({
+      headers: {
+        Authorization: 'Bearer some token',
+        'Content-Type': 'application/json',
+      },
+      method: 'PUT',
+      body: JSON.stringify(objectToUpdate),
+    });
+  });
+
+  it('resolves with a 404 and handles it when it fails to update the resource (api)', async () => {
+    mockAppData = { jwt: 'some token' };
+    const objectToUpdate = { object: 'data' };
+    fetchMock.mock('/api/model-name/1/', 404);
+
+    let thrownError;
+
+    try {
+      await updateOne({
+        name: 'model-name',
+        id: '1',
+        object: objectToUpdate,
+      });
+    } catch (error) {
+      thrownError = error;
+    }
+
+    expect(thrownError).toEqual({ code: 'exception' });
+
+    expect(fetchMock.lastCall()![0]).toEqual('/api/model-name/1/');
+    expect(fetchMock.lastCall()![1]).toEqual({
+      headers: {
+        Authorization: 'Bearer some token',
+        'Content-Type': 'application/json',
+      },
+      method: 'PUT',
+      body: JSON.stringify(objectToUpdate),
+    });
+  });
+
+  it('resolves with a 400 and handles it when it fails to update the resource (api)', async () => {
+    mockAppData = { jwt: 'some token' };
+    const objectToUpdate = { object: 'data' };
+    fetchMock.mock('/api/model-name/1/', 400);
+
+    await expect(
+      updateOne({
+        name: 'model-name',
+        id: '1',
+        object: objectToUpdate,
+      }),
+    ).rejects.toThrow(
+      'invalid json response body at /api/model-name/1/ reason: Unexpected end of JSON input',
```

Comment:
I don't understand how you have this error message.  It's a kind of black magic for me. Can you explain me ? Because it doesn't match what is thrown in the `updateOne` code. An object is thrown and I don't see any "serailization"

Reply:
To be honest, it's an adaptation from  createOne.
As it's works like this for creating resource, and in both cases the related object is send from API, I think it should be similar.
But @mbenadda surely can answer your question, or correct my code.

Reply:
By my understanding : as we mock fetch, by returning a 400 with an empty body, the string in test is build with those lines

    if (response.status === 400) {
      throw { code: 'invalid', ...(await response.json()) };
    }

Reply:
Ok so I understand what is happening and your test is wrong. It fails because `response.json()` fails, there is no valid json in the body response so calling `await reponse.json` fails with the message `invalid json response` etc.

So you have to fix this test and the one in the `createOne` tests.

Thanks.

Reply:
I think a 400 error from the server should have proper JSON with data on the error so this is the way to go!

Reply:
> I think a 400 error from the server should have proper JSON with data on the error so this is the way to go!

Yes but the test is wrong. The response body is null and should have a valid json.

Reply:
Indeed I was just agreeing with you 👍